os/drivers/lwnl/lwnl.c

```diff
@@ -204,7 +212,7 @@ static int lwnl_ioctl(struct file *filep, int cmd, unsigned long arg)
 	LWNL_ENTER;
 	int res = lwnl_add_listener(filep);
 	if (res < 0) {
-		res = -ENOSYS;
+		res = -EBADF;
```

Comment:
In this function, there are two cases where the return type is set to '-EBADF'. How could you tell them apart?

Reply:
According to ioctl manual page. ioctl can return following errno values.
---
EBADF  fd is not a valid file descriptor.

       EFAULT argp references an inaccessible memory area.

       EINVAL request or argp is not valid.

       ENOTTY fd is not associated with a character special device.

       ENOTTY The specified request does not apply to the kind of object that the file descriptor fd references
---
There is no relevant errno value for this. If you have good idea, I'll apply it.